prototype/views.py

```diff
@@ -0,0 +1,43 @@
+from django.urls import reverse
+
+from directory_components.helpers import SocialLinkBuilder
+
+from prototype.mixins import CMSPageViewMixin, SlugFromKwargsMixin
+
+
+class NewsPageView(CMSPageViewMixin):
+    template_name = 'prototype/news_list.html'
+    slug = 'news'
+
+
+class TopicListPageView(CMSPageViewMixin, SlugFromKwargsMixin):
+    template_name = 'prototype/topic_list.html'
+
+
+class ArticleListPageView(CMSPageViewMixin, SlugFromKwargsMixin):
+    template_name = 'prototype/article_list.html'
+
+
+class ArticleDetailView(CMSPageViewMixin, SlugFromKwargsMixin):
+    template_name = 'prototype/article_detail.html'
+    url_name = 'article-detail'
+
+    def get_context_data(self, *args, **kwargs):
+
+        social_links_builder = SocialLinkBuilder(
+            self.request.build_absolute_uri(
+                reverse(self.url_name, kwargs={**self.kwargs})
```

Comment:
`reverse(self.url_name, kwargs={**self.kwargs})` can be replaced with `self.request.path`

Reply:
oh! in fact I think this we can just use `self.request.build_absolute_uri()` without any path provided :)

Reply:
none of those work unfortunately :(